File: 1536-minimum-swaps-to-arrange-a-binary-grid/1536-minimum-swaps-to-arrange-a-binary-grid.cpp
class Solution {
public:
    int minSwaps(vector<vector<int>>& grid) {
        int res = 0;
        int n = grid.size();
        vector<int> pos(n, -1);
        for(int i = 0; i < n; i++)
        {
            for(int j = n - 1; j >= 0; j--)
            {
                if(grid[i][j] == 1)
                {
                    pos[i] = j;
                    break;
                }
            }
        }
        for(int i = 0; i < n; i++)
        {
            bool found = false;
            for(int j = i; j < n; j++)
            {
                if(pos[j] <= i)
                {
                    found = true;
                    for(int k = j; k > i; k--)
                    {
                        swap(pos[k], pos[k - 1]);
                        res++;
                    }
                    break;
                }
            }
            if(!found)
            {
                return -1;
            }
        }
        return res;
    }
};

File: 1536-minimum-swaps-to-arrange-a-binary-grid/1536-minimum-swaps-to-arrange-a-binary-grid.py
class Solution:
    def minSwaps(self, grid: List[List[int]]) -> int:
        res = 0
        pos = [-1 for _ in range(len(grid))]
        for i in range(len(grid)):
            for j in range(len(grid) - 1, -1, -1):
                if grid[i][j] == 1:
                    pos[i] = j
                    break
        for i in range(len(grid)):
            found = False
            for j in range(i, len(grid)):
                if pos[j] <= i:
                    found = True
                    for k in range(j, i, -1):
                        pos[k], pos[k - 1] = pos[k - 1], pos[k]
                        res += 1
                    break
            if not found:
                return -1
        return res
